### PR TITLE
feat(sbb-tooltip): add sbb-tooltip-trigger

### LINKED ISSUE #1478: story(sbb-tooltip): provide sbb-tooltip-trigger
### Feature Description

We want to provide a tooltip trigger component to improve keyboard navigation and also accessibility. For this purpose, we want to implement `sbb-tooltip-trigger`, which is mainly a transparent button which adapts to its child size.

[Figma Spec](https://www.figma.com/file/9r6xSfNmEfCFxl1yFYedrj/Lyne-Components?node-id=15414%3A517130&t=araOSZUOlpHffACB-1)

**Design Spec**
- The button should adapt its size to the content when used inline in text, start with default size of icons.
- Color varies in different states.

**Technical Spec**
- Use `buttonReset` sass mixin.
- Tooltip should be opened by pressing space or enter.
- @Prop iconName = 'circle-information-small'; ('i' icon should be default);
- Provide unnamed slot to enable custom content, replaces the icon prop.
- Update stories in tooltip with trigger component.

```html
<sbb-tooltip-trigger id="tooltip-trigger"></sbb-tooltip-trigger>

<sbb-tooltip trigger="tooltip-trigger">
  <p>
    Tooltip content. 
  </p>
</sbb-tooltip>
```

```html
<sbb-tooltip-trigger id="tooltip-trigger2">
  Custom Content
</sbb-tooltip-trigger>

<sbb-tooltip trigger="tooltip-trigger2">
  <p>
    Tooltip content. 
  </p>
</sbb-tooltip>
```

### Definition of Done

- [x] Component is implemented
- [x] Documentation is provided
- [x] Tests are implemented
- [x] E2E Tests are implemented
- [x] Storybook stories are implemented
- [x] Navigation via keyboard is tested
- [ ] Screen reader output is tested
- [x] High-contrast is tested
- [ ] Remaining accessibility is tested
- [ ] UX approved

### PR DESCRIPTION
**Pull request by simone-sabato**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue 

This PR Closes #1478

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- provide `sbb-tooltip-trigger`

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [x] Edge Desktop
- [x] Safari Desktop
- [x] Chrome Mobile
- [x] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [x] VoiceOver Safari Desktop
- [x] VoiceOver Chrome Desktop
- [x] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+:host {
+  --sbb-tooltip-background-hover: #444;
+  --sbb-tooltip-background-active: #5a5a5a;
+
+  display: inline-block;
+}
+
+.sbb-tooltip-trigger {
+  @include sbb.button-reset;
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.scss:15]: We should inherit the text styles if it's embedded into a text. Maybe it's necessary to also include this text-inherit mixin on host level. Please test the scenario if there is a parent element having a custom font-family, font-weight, font-size, color, line-height, others? and see how the tooltip trigger reacts. Take the custom content story for that.

```suggestion
  @include sbb.button-reset;
  @include sbb.text-inherit;
```

File: src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.stories.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+const tooltip = () => (
+  <sbb-tooltip data-testid="tooltip" trigger="tooltip-trigger">
+    <p id="tooltip-content" style={'margin: 0; font-size: var(--sbb-font-size-text-s);'}>
+      Simple information tooltip with link.
+    </p>
+  </sbb-tooltip>
+);
+
+const Template = (args) => [<sbb-tooltip-trigger {...args}></sbb-tooltip-trigger>, tooltip()];
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.stories.js:34]: Please wrap Text around tooltip trigger and make sure that it aligns correctly with the text.

NOK: 
image

File: src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.custom.d.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface InterfaceSbbTooltipTriggerAttributes {
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.custom.d.ts:1]: removed unused file

File: src/components/sbb-tooltip-trigger/readme.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# sbb-tooltip-trigger
+The `sbb-tooltip-trigger` component is used to trigger a tooltip by pressing a button. By default is rendered a `sbb-icon`. It is possible to override the `sbb-icon` with custom contents.
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip-trigger/readme.md:2]:
```suggestion
The `sbb-tooltip-trigger` component is used to trigger a tooltip by pressing a button. By default it renders the `sbb-icon` `information-circle-small`. With the property `icon-name` it is possible to define a custom icon. Moreover, it is possible to override the `sbb-icon` with custom content by slotting it.
```

File: src/components/sbb-tooltip-trigger/readme.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+## Usage
+
+The examples below shows how to use the component with the `<sbb-icon>` and custom content.
+ 
+ ```html
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip-trigger/readme.md:8]: Please use a ```html for each example with a short title.

File: src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.e2e.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import events from '../sbb-tooltip/sbb-tooltip.events';
+import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
+
+describe('sbb-tooltip-trigger', () => {
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.e2e.ts:4]: Nice tests :)

File: src/components/sbb-tooltip/sbb-tooltip.tsx [hunk 7]
```diff
@@ -352,10 +352,12 @@ export class SbbTooltip implements ComponentInterface {
   // Set focus on the first focusable element.
   private _setTooltipFocus(): void {
     this._prevFocusedElement = document.activeElement as HTMLElement;
-    this._firstFocusable = this._element.querySelector(IS_FOCUSABLE_QUERY);
+    this._firstFocusable =
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip/sbb-tooltip.tsx:355]: Why was this change necessary?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
- You accidentally removed documentations of several components. Please revert. I recommend to always have a look at the changes after creating a PR.
- Please migrate the current tooltip examples to the new tooltip-trigger (Contact @dauriamarco if needed)
- Please check accessibility and especially if there would be an accessibilityLabel needed. Reach out to @dauriamarco as well, if needed.

File: src/components/sbb-tooltip/sbb-tooltip.tsx (hunk 7 above)
Comment (jeripeierSBB) [reply src/components/sbb-tooltip/sbb-tooltip.tsx:355]: @dauriamarco : we did this because the first focusable element might be present in the content provided by the consumer, otherwise it's the close button (inside the shadowRoot)

Comment (dauriamarco) [reply src/components/sbb-tooltip/sbb-tooltip.tsx:355]: Thank you for quoting me 😄. I also have an additional consideration: I'm not sure whether just using `this._element.querySelector(IS_FOCUSABLE_QUERY)` is enough or if we should use the `getFocusableElements` function from the `focus.ts` file. What do you think?

Comment (kyubisation) [reply src/components/sbb-tooltip/sbb-tooltip.tsx:355]: I think `IS_FOCUSABLE_QUERY` is enough in this use case, as we are primarily interested in the light DOM elements. Maybe it might make sense to directly query for the close button?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

File: src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.stories.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  <sbb-tooltip data-testid="tooltip" trigger="tooltip-trigger">
+    <span id="tooltip-content" class="sbb-text-s">
+      Simple information tooltip with link.
+    </span>
+  </sbb-tooltip>
+);
+
+const Template = (args) => [
+  <span class="sbb-text-s" style="display: flex; align-items: center">
+    <span style="margin-inline-end: var(--sbb-spacing-fixed-1x)">This is a demo text.</span>
```
Comment (jeripeierSBB) [inline src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.stories.js:36]: I'm not sure if this margin inline end is really wanted by design. @mcilurzo Can you give a feedback here, or make some examples also in Figma? Also should a dot be before or after the trigger?

Comment (jeripeierSBB) [reply src/components/sbb-tooltip-trigger/sbb-tooltip-trigger.stories.js:36]: Margin is wanted, just vertically should be centered also if icon is smaller than line-height (display flex on button probably)

**Review by jeripeierSBB — APPROVED** [review]
LGTM, thanks
